### PR TITLE
Add test cases for soft constraints/multi constraints

### PR DESCRIPTION
**Pull request by damemi**
Fixes https://github.com/kubernetes-sigs/descheduler/issues/564
/kind bug

### CODE REVIEW

File: pkg/descheduler/strategies/topologyspreadconstraint.go [hunk 1]
```diff
@@ -126,7 +126,7 @@ func RemovePodsViolatingTopologySpreadConstraint(
 		for _, pod := range namespacePods.Items {
 			for _, constraint := range pod.Spec.TopologySpreadConstraints {
 				// Ignore soft topology constraints if they are not included
-				if (strategy.Params == nil || !strategy.Params.IncludeSoftConstraints) && constraint.WhenUnsatisfiable != v1.DoNotSchedule {
+				if constraint.WhenUnsatisfiable == v1.ScheduleAnyway && (strategy.Params == nil || !strategy.Params.IncludeSoftConstraints) {
```
Comment (damemi) [inline pkg/descheduler/strategies/topologyspreadconstraint.go:129]: changed this to be a bit more readable

**Comment by damemi** [comment]
Turns out there wasn't actually a bug with this, but we should still have these test cases anyway

File: pkg/descheduler/strategies/topologyspreadconstraint_test.go [hunk 2]
```diff
@@ -96,6 +96,41 @@ func TestTopologySpreadConstraint(t *testing.T) {
...
+					labels: map[string]string{"foo": "bar"},
+				},
+			}),
+			expectedEvictedCount: 1,
+			strategy:             api.DeschedulerStrategy{Params: &api.StrategyParameters{IncludeSoftConstraints: true}},
+			namespaces:           []string{"ns1"},
+		},
 		{
 			name: "NEW",
 			//name: "2 domains, sizes [3,1], maxSkew=1, no pods eligible, move 0 pods",
```
Comment (lixiang233) [inline pkg/descheduler/strategies/topologyspreadconstraint_test.go:136]: Can you update the name of this testcase by the way?

File: pkg/descheduler/strategies/topologyspreadconstraint.go (hunk 1 above)
Comment (ingvagabund) [reply pkg/descheduler/strategies/topologyspreadconstraint.go:128]: +1 for inclusion, not exclusion

File: pkg/descheduler/strategies/topologyspreadconstraint_test.go (hunk 2 above)
Comment (damemi) [reply pkg/descheduler/strategies/topologyspreadconstraint_test.go]: Ah whoops, thanks for catching that!

**Pull request merged**